### PR TITLE
HTTP logger for APISIX 

### LINKED ISSUE #965: HTTP logger to send logs to external log management tool
Does  APISIX provide an HTTP logger to send gateway request (url, uri, time, key) logs to an external endpoint? I want to send the access logs to [Kibana/Logstash](https://www.elastic.co/what-is/elk-stack) for further analysis.

### PR DESCRIPTION
**Pull request by sshniro**
### Summary
Resolves #965 

Using the [lua-resty-http](https://github.com/ledgetech/lua-resty-http) package and also utilizing the `genericinterface` to configure the `set_keepalive` method.

### CODE REVIEW

**Comment by sshniro** [comment]
Hi @membphis  I'm using openresty `socket.url` to decode the URL but getting this message in the test case.

```lua
local url = require "socket.url"
local function test(conf){
local url_decoded = socket.url.parse(conf.uri)
-- host, port etc 
}
```

In this case, shall I use this library to decode the URL? : https://github.com/golgote/neturl

**Comment by sshniro** [comment]
Hi @membphis , @moonming the logger is ready for review.

**Comment by sshniro** [comment]
@membphis can you have a look at this PR as well?

**Comment by sshniro** [comment]
@membphis , @moonming  any update on this? I would love to use this plugin in my prod setup with Kibana.

File: apisix/plugins/http-logger.lua [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+-- Unless required by applicable law or agreed to in writing, software
+-- distributed under the License is distributed on an "AS IS" BASIS,
+-- WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+-- See the License for the specific language governing permissions and
+-- limitations under the License.
+--
+local core = require("apisix.core")
+local log_util = require("apisix.utils.log-util")
+local batch_processor = require("apisix.utils.batch-processor")
+local plugin_name = "udp-logger"
```
Comment (moonming) [inline apisix/plugins/http-logger.lua:20]: typo?

Comment (sshniro) [reply apisix/plugins/http-logger.lua]: fixed.

**Comment by sshniro** [comment]
It seems to be a temporary issue with the build process, need to restart it from Travis.
```
gunzip failed: input file '/Users/travis/.cpanm/sources/http%www.cpan.org/02packages.details.txt.gz' does not exist
 at /usr/local/bin/cpanm line 462.
+exit 1
The command "$PWD/.travis/${OSNAME}_runner.sh before_install" failed and exited with 1 during .

```

**Comment by sshniro** [comment]
@membphis the conflicts have been resolved.

**Comment by sshniro** [comment]
@membphis @moonming if the code is okay, shall we merge this PR?

**Comment by moonming** [comment]
merged, thx @sshniro

**Comment by sshniro** [comment]
Nice, thanks :)

**Pull request merged**